docs/contributors/code/testing-overview.md

```diff
@@ -212,6 +212,91 @@ describe( 'my module', () => {
 } );
 ```
 
+### User interactions
+
+Simulating user interactions is a great way to **write tests from the user's perspective**, and therefore avoid testing implementation details.
+
+When writing tests with Testing Library, there are two main alternatives for simulating user interactions:
+
+1. The [`fireEvent`](https://testing-library.com/docs/dom-testing-library/api-events/#fireevent) API, a utility for firing DOM events part of the Testing Library core API.
+2. The [`user-event`](https://testing-library.com/docs/user-event/intro/) library, a companion library to Testing Library that simulates user interactions by dispatching the events that would happen if the interaction took place in a browser.
+
+The built-in `fireEvent` is a utility for dispatching DOM events. It dispatches exactly the events that are described in the test spec - even if those exact events never had been dispatched in a real interaction in a browser.
+
+On the other hand, the `user-event` library exposes higher-level methods (e.g. `type`, `selectOptions`, `clear`, `doubleClick`...), that dispatch events like they would happen if a user interacted with the document, and take care of any react-specific quirks.
+
+For the above reasons, **the `user-event` library is the recommended way to simulate user events when writing tests**.
+
+**Not so good**: using `fireEvent` to dispatch DOM events.
+
+```javascript
+import { render, screen } from '@testing-library/react';
+
+test('fires onChange when a new value is typed', () => {
+	const spyOnChange = jest.fn();
+
+	// A component with one `input` and one `select`.
+	const { user } = render(<MyComponent onChange={spyOnChange} />)
+
+	const input = screen.getByRole( 'textbox' );
+	input.focus();
+	// No clicks, no key events.
+	fireEvent.change( input, { target: { value: 62 } } );
+
+	// The `onChange` callback gets called once with '62' as the argument.
+	expect( setState ).toHaveBeenCalledTimes( 1 );
+
+	const select = screen.getByRole( 'listbox' );
+	select.focus();
+	// No pointer events dispatched.
+	fireEvent.change( select, { target: { value: 'optionValue' } } );
+
+	// ...
+```
+
+**Good**: using `user-event` to simulate user events.
+
+```javascript
+
+import { render as RTLrender, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+function render(jsx) {
+	return {
+		user: userEvent.setup(),
+		...RTLrender(jsx),
+	}
+}
```

Comment:
Hmm, I think I prefer the inlined version (`const user = userEvent.setup()`) instead of this setup abstraction... 🤔 It's just one line, and I kind of find it more simple and self-explanatory. What do you think? Maybe some idioms will emerge once v14 gets widespread adoption.

https://testing-library.com/docs/user-event/intro#writing-tests-with-userevent